tests/utils/skipif.php

```diff
@@ -15,11 +15,17 @@
  */
 function disable_skipif_caching()
 {
+    static $skipifCachingDisabled;
```

Comment:
> While this can lead to skip messages containing an extra nocache at the end, I believe this is preferable to SKIPIF logic being ignored

I assume that was written before you added this `static` sentinel value? I don't think we need to worry about duplicate "nocache" lines.

Reply:
Not quite. Assuming one check exits with the message "skip this test does not work", the shutdown handler would still add "nocache" at the end, changing the message to "skip this test does not worknocache".

Reply:
Is it necessary to append `nocache` to an existing line to avoid a BORK if it were printed on its own line after a `skip`?

Sorry, it's been quite a while since I thought about any of this.

Reply:
No, it's more of a side-effect of how things are done: our skip functions call `exit('skip whatever');` without a trailing newline. The shutdown handler then prints "nocache" to the output, meaning that it will be appended to the end of whatever the previous skip functions printed. As the message doesn't have a trailing newline, it's appended in the same line.

Looking at run-tests.php again, I can see that the output of the skip file is run through `trim` before checking for "nocache", so we could even print a leading newline in the shutdown handler to avoid appending to a previously printed skip message. I'll take another look at that.